be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -27,6 +33,11 @@ object ElectionValidator extends MessageDataContentValidator with EventValidator
           Right(validationError(s"'end_time' (${data.end_time}) timestamp is smaller than 'start_time' (${data.start_time})"))
         } else if (expectedHash != data.id) {
           Right(validationError("unexpected id"))
+        } else if (!validateOwner(sender, channel)){
+          Right(validationError(s"invalid sender $sender"))
+        } //note: the SetupElection is the only message sent to the main channel, others are sent in an election channel
+        else if (!validateChannelType(ObjectType.LAO, channel)) {
+          Right(validationError(s"trying to send a CloseRollCall message on a wrong type of channel $channel"))
```

Comment:
Did you mean "trying to send a SetupElection message on a wrong type of channel $channel"

Reply:
yeah I did